internal/commands/manifest_push_test.go

```diff
@@ -0,0 +1,89 @@
+package commands_test
+
+import (
+	"bytes"
+	"errors"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/heroku/color"
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+	"github.com/spf13/cobra"
+
+	"github.com/buildpacks/pack/internal/commands"
+	"github.com/buildpacks/pack/internal/commands/testmocks"
+	"github.com/buildpacks/pack/pkg/logging"
+	h "github.com/buildpacks/pack/testhelpers"
+)
+
+func TestManifestPushCommand(t *testing.T) {
+	color.Disable(true)
+	defer color.Disable(false)
+
+	spec.Run(t, "Commands", testManifestPushCommand, spec.Random(), spec.Report(report.Terminal{}))
+}
+
+func testManifestPushCommand(t *testing.T, when spec.G, it spec.S) {
+	var (
+		command        *cobra.Command
+		logger         *logging.LogWithWriters
+		outBuf         bytes.Buffer
+		mockController *gomock.Controller
+		mockClient     *testmocks.MockPackClient
+	)
+
+	it.Before(func() {
+		logger = logging.NewLogWithWriters(&outBuf, &outBuf)
+		mockController = gomock.NewController(t)
+		mockClient = testmocks.NewMockPackClient(mockController)
+
+		command = commands.ManifestPush(logger, mockClient)
+	})
+	it("should annotate images with given flags", func() {
+		preparePushManifest(t, mockClient)
+
+		command.SetArgs([]string{
+			"some-index",
+			"-f",
+			"docker",
+			"--purge",
+			"--insecure",
+		})
+		h.AssertNil(t, command.Execute())
```

Comment:
Ideally we would also assert that the command options are passed through to the client. This would require some re-working of `preparePushManifest` as the expectations are defined there.